File: text-zoom/ios/Plugin/TextZoomPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(TextZoomPlugin, "TextZoom",
           CAP_PLUGIN_METHOD(getPreferred, CAPPluginReturnPromise);
)

File: text-zoom/ios/Plugin/TextZoomPlugin.swift
import Foundation
import Capacitor

@objc(TextZoomPlugin)
public class TextZoomPlugin: CAPPlugin {
    private let textZoom = TextZoom()

    @objc func getPreferred(_ call: CAPPluginCall) {
        call.resolve([
            "value": textZoom.preferredFontSize()
        ])
    }

    @objc func get(_ call: CAPPluginCall) {
        call.unimplemented("Not available on iOS")
    }

    @objc func set(_ call: CAPPluginCall) {
        call.unimplemented("Not available on iOS")
    }
}
